numpy/core/src/npysort/quicksort.c.src

```diff
@@ -107,6 +111,7 @@ quicksort_@suff@(void *start, npy_intp num, void *NOT_USED)
                 *sptr++ = pi - 1;
                 pl = pi + 1;
             }
+            *psdepth++ = cdepth + 1;
```

Comment:
I'm pretty sure it is wrong. Every push on the stack should increment, depth has nothing to do with actual stack depth here.